python/paddle/static/io.py

```diff
@@ -979,6 +1014,32 @@ def load_inference_model(path_prefix, executor, **kwargs):
     return [program, feed_target_names, fetch_targets]
 
 
+@overload
+@dygraph_not_support
+def save_vars(
+    executor: Executor,
+    dirname: None,
+    main_program: Program | None = None,
+    vars: list[Tensor] | None = None,
+    predicate: callable[[Tensor], bool] | None = None,
+    filename: None = None,
+) -> None:
+    ...
+
+
+@overload
+@dygraph_not_support
+def save_vars(
+    executor: Executor,
+    dirname: str,
+    main_program: Program | None = None,
+    vars: list[Tensor] | None = None,
+    predicate: callable[[Tensor], bool] | None = None,
```

Comment:
```suggestion
    predicate: Callable[[Tensor], bool] | None = ...,
```

`overload` 中不需要再赋默认值 ～